File: application/src/main/java/de/wuespace/telestion/project/daedalus2/redis/RedisTimeSeriesRequest.java
package de.wuespace.telestion.project.daedalus2.redis;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.mongodb.lang.NonNull;
import de.wuespace.telestion.api.message.JsonMessage;

import java.util.List;

@SuppressWarnings("unused")
public record RedisTimeSeriesRequest(
		@NonNull
		@JsonProperty List<RedisTimeSeriesSpecification> fields
) implements JsonMessage {
	public RedisTimeSeriesRequest() {
		this(List.of());
	}
}

File: client/src/widgets/hooks/redis/model/redis-time-series-request.ts
import { RedisTimeSeriesSpecification } from './redis-time-series-specification';
import { JavaMessage } from '../../../../model/java-message';

/**
 * Counterpart to `de.wuespace.telestion.project.daedalus2.redis.RedisTimeSeriesRequest` Java class
 */
export interface RedisTimeSeriesRequest extends JavaMessage {
	fields: RedisTimeSeriesSpecification[];
	className: 'de.wuespace.telestion.project.daedalus2.redis.RedisTimeSeriesRequest';
}
